src/dataflow-types/src/client.rs

```diff
@@ -658,6 +667,120 @@ pub mod partitioned {
 
     use timely::progress::frontier::MutableAntichain;
 
+    /// Buffer for partial results for a `TAIL`.
+    /// This exists so we can consolidate rows and
+    /// sort them by timestamp before passing them to the consumer
+    /// (currently, coord.rs).
+    struct PendingTail {
+        /// `frontiers[i]` is an antichain representing the times at which we may
+        /// still receieve results from worker `i`.
+        frontiers: HashMap<usize, Antichain<Timestamp>>,
```

Comment:
`Vec<Antichain<Timestamp>>` initialized with the min element? We know that Timely workers are numbered 0..workers-1.

Reply:
I think the `HashMap` is better. The same `partitioned` implementation is used at each of the deployed processes, and the workers they hold will not always start at zero.